extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -172,6 +172,24 @@ private Uni<TenantConfigContext> createTenantContext(Vertx vertx, OidcTenantConf
             return Uni.createFrom().failure(t);
         }
 
+        if (oidcConfig.roles.source.orElse(null) == Source.userinfo
+                || oidcConfig.token.verifyAccessTokenWithUserInfo
+                || !oidcConfig.authentication.isIdTokenRequired().orElse(true)) {
+
+            Optional<Boolean> userInfoRequired = oidcConfig.authentication.isUserInfoRequired();
+            if (userInfoRequired.isPresent()) {
+                if (!userInfoRequired.get()) {
+                    throw new ConfigurationException(
+                            "UserInfo is not required but one of the following conditions is true: "
```

Comment:
Shouldn't this error message contain line breaks?

Reply:
Also I wonder if it's possible to be more objective, as saying `one of the following conditions is true` is like saying "we haven't checked what happened but one of these things is true, guess what?" 😀

Reply:
@gastaldi Hi, yes, I've been a bit lazy I guess, wanted to avoid too many checks, agreed, needs to be more specific, thanks

Reply:
@gastaldi Does it look a bit better now :-) ?